test/truncate_test.go

```diff
@@ -56,3 +59,54 @@ func TestSimpleTruncate(t *testing.T) {
 		),
 	)
 }
+
+func testTruncateLargeThenWriteToSmallerOffset(t *testing.T, dataLen int) {
+	data := make([]byte, dataLen)
+	for i := 0; i < len(data); i++ {
+		if i < 12 || (i >= 64 && i < 68) || (i >= dataLen-12) {
+			data[i] = byte(i)
```

Comment:
Technically you're truncating `i` for large values (used in `TestTruncateLargeThenWriteToSmallerOffsetWithHoles`), but I don't think it matters.